doc/2/api/controllers/document/delete-by-query/index.md

```diff
@@ -14,7 +14,13 @@ Documents removed that way trigger real-time notifications.
 
 The request fails if the number of documents returned by the search query exceeds the `documentsWriteCount` server configuration (see the [Configuring Kuzzle](/core/2/guides/essentials/configuration) guide).
 
-To remove a greater number of documents, either change the server configuration, or split the search query.
+This behavior is to prevent Kuzzle from Out of Memory error or DOS because every deleted document has to be fetched first to trigger realtime notifications.
```

Comment:
I don't think using `deleteByQuery` can be considered a DoS attack.

```suggestion
This behavior aims at limiting the pressure on memory and on real-time notifications.
```

Reply:
I meant that the OOM can lead to DOS but I agree it could be confusing